plugins/modules/discovery_intent.py

```diff
@@ -475,9 +489,11 @@ def preprocess_device_discovery(self, ip_address_list=None):
             else:
                 self.preprocess_device_discovery_handle_error()
         elif discovery_type == "CIDR":
-            if len(ip_address_list) == 1 and self.validated_config[0].get('prefix_length'):
-                ip_address_list = ip_address_list[0]
-                ip_address_list = str(ip_address_list) + "/" + str(self.validated_config[0].get('prefix_length'))
+            if len(ip_address_list) == 1:
+                if len(ip_address_list[0].split("/")) == 2:
+                    ip_address_list = ip_address_list[0]
+                else:
+                    ip_address_list = str(ip_address_list[0]) + "/" + str("30")
```

Comment:
Can we give a warning message? Say that prefix length is not given, hence taking 30 as default..

Reply:
Done